ui/src/main/webapp/app/graph/graph-component.js

```diff
@@ -630,9 +654,24 @@ function GraphController($q, graph, config, error, loading, query, operationOpti
      * Removes every selected element in the graph.
      */
     vm.removeSelected = function() {
-        cytoscapeGraph.filter(":selected").remove();
+        var removedElements = cytoscapeGraph.filter(":selected").remove();
+        for(var i = 0; i < removedElements.length; i++) {
+            graph.getRemovedElements().push(removedElements[i]);
+        }
+
         cytoscapeGraph.elements().unselect();
         vm.selectedElements.entities = [];
         vm.selectedElements.edges = [];
     }
+
+    var removePreviouslyRemovedElements = function() {
+        var elements = cytoscapeGraph.elements();
+        for(var i = 0; i < graph.getRemovedElements().length; i++) {
```

Comment:
Could you use a for each loop here? https://devdocs.io/javascript/statements/for...of

Reply:
Unfortunately elements is a funny object and if I try and use that I get this error:
```Uncaught TypeError: elements is not iterable```

Reply:
You could use the [eles.forEach method](https://js.cytoscape.org/#eles.forEach)